editor/plugins/material_editor_plugin.h

```diff
@@ -84,29 +98,37 @@ class MaterialEditor : public Control {
 	Button *quad_switch = nullptr;
 	Button *light_1_switch = nullptr;
 	Button *light_2_switch = nullptr;
+	Button *floor_switch = nullptr;
+
+	Shape shape;
+
+	Vector2 rot;
+	float cam_zoom = 3.0f;
+	AABB contents_aabb;
+
+	Ref<BaseMaterial3D> default_floor_material;
 
 	struct ThemeCache {
 		Ref<Texture2D> light_1_icon;
 		Ref<Texture2D> light_2_icon;
+		Ref<Texture2D> floor_icon;
 		Ref<Texture2D> sphere_icon;
 		Ref<Texture2D> box_icon;
 		Ref<Texture2D> quad_icon;
 		Ref<Texture2D> checkerboard;
 	} theme_cache;
 
-	void _on_light_1_switch_pressed();
-	void _on_light_2_switch_pressed();
-	void _on_sphere_switch_pressed();
-	void _on_box_switch_pressed();
-	void _on_quad_switch_pressed();
+	void _on_visibility_switch_pressed(const int &p_shape);
+	void _on_shape_switch_pressed(const int &p_shape);
```

Comment:
```suggestion
	void _on_visibility_switch_pressed(int p_shape);
	void _on_shape_switch_pressed(int p_shape);
```
Don't use const reference with primitive types.